drakrun/drakrun/main.py

```diff
@@ -265,18 +275,22 @@ def process(self):
             self.log.info("note that artifacts will be stored under this overriden identifier")
 
         workdir = '/tmp/drakrun/vm-{}'.format(int(INSTANCE_ID))
+
         extension = self.current_task.headers.get("extension", "exe").lower()
-        start_command = 'start malwar.{}'.format(extension)
+        if '(DLL)' in magic_output:
+            extension = 'dll'
+        self.log.info("Running file as  %s", extension)
 
-        if extension == 'dll' or '(DLL)' in magic_output:
-            run_command = self._get_dll_run_command(sample.content)
+        file_name = self.current_task.payload.get("file_name", f"malwar.{extension}")
+        self.log.info("Using file name %s", file_name)
 
-            if not run_command:
-                self.log.info('Unable to run malware sample, could not generate any suitable command to run it.')
-                return
+        start_command = self.current_task.payload.get("start_command", self._get_start_command(extension, sample))
+        if not start_command:
+            self.log.error('Unable to run malware sample, could not generate any suitable command to run it.')
+            return
 
-            self.log.info('Using command: {}'.format(run_command))
-            start_command = 'start {}'.format(run_command)
+        start_command = start_command.replace("%f", file_name)
```

Comment:
I hope this doesn't conflict with anything in the future. For now it seems to be fair choice.